src/extension.ts

```diff
@@ -3,34 +3,54 @@
 *  Licensed under the MIT License.
 *--------------------------------------------------------------------------------------------*/
 
-import * as languageclient from 'vscode-languageclient';
-import * as logger from './logger';
 import * as path from 'path';
-import * as schemacontributor from './schema-contributor';
 import * as vscode from 'vscode';
-import * as schemaassociationservice from './schema-association-service';
-import TelemetryReporter from 'vscode-extension-telemetry';
-const fs = require('fs');
+import { createTelemetryReporter, callWithTelemetryAndErrorHandling, IActionContext, AzureUserInput, registerUIExtensionVariables } from 'vscode-azureextensionui';
+import * as languageclient from 'vscode-languageclient';
+
 import { activateConfigurePipeline } from './configure/activate';
+import { extensionVariables } from './configure/model/models';
+import * as logger from './logger';
+import * as schemaassociationservice from './schema-association-service';
+import * as schemacontributor from './schema-contributor';
 
-const myExtensionId = 'azure-pipelines';
-const telemetryVersion = generateVersionString(vscode.extensions.getExtension(`ms-azure-devops.${myExtensionId}`));
-const telemetryKey = 'ae672644-d394-497c-8c57-98f6eac32342';
+let perfStats = {
+    loadStartTime: Date.now(),
+    loadEndTime: undefined
+};
 
-let reporter: TelemetryReporter;
+const configurePipelineEnabled: boolean = vscode.workspace.getConfiguration('[azure-pipelines]', null).get('configure') ? true : false;
```

Comment:
can this be user level setting as well?

Reply:
I didn't really understand this ?

Reply:
I guess settings in vs code are at two levels - user settings and workspace settings. My question is whether we need to look at both type of settings

Reply:
When we call this function to get the configuration, the vscode takes care of returning back the right configuration. Hence we do not need to take care of this explicitly. 
They must have rules inside SDK: such as workspace setting override user settings.